### PR TITLE
Adding fields and blocks classes

### PR DESCRIPTION
**Pull request by SNsir**

### CODE REVIEW

File: galite-core/src/main/kotlin/org/kopi/galite/form/VIntegerField.kt [hunk 1]
```diff
@@ -15,6 +15,370 @@
...
+   */
+  fun getInt(r: Int): Int = getObject(r) as Int
+
+  /**
+   * Returns the field value of the current record as an object
+   */
+  override fun getObjectImpl(r: Int): Any? = value[r]
+
+  override fun toText(o: Any?): String {
+    return if (o == null) "" else (o as Int?).toString()
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VIntegerField.kt:204]: can we change r (o as Int?).toString()   --> o.toString()

File: galite-core/src/main/kotlin/org/kopi/galite/form/VIntegerField.kt [hunk 2]
```diff
@@ -15,6 +15,370 @@
...
+    this.criticalMaxValue = criticalMaxValue
+  }
+
+  fun setCriticalMinValue(criticalMinValue: Int?) {
+    this.criticalMinValue = criticalMinValue
+  }
+
+  fun setCriticalMaxValue(criticalMaxValue: Int?) {
+    this.criticalMaxValue = criticalMaxValue
+  }
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VIntegerField.kt:355]: can we delete them then change criticalMaxValue & criticalMinValue to public !

File: galite-core/src/main/kotlin/org/kopi/galite/form/ViewBlockAlignment.kt [hunk 3]
```diff
@@ -15,6 +15,51 @@
...
+
+    x-- // we want to align middle
+    target = align.getTargetAt(x)
+    val view: UBlock? = formView.getBlockView(align.getBlock())
+
+    if (target != -1) {
+      if (view == null) {
+        return 0
+      }
+      val pos = if (isChart()) target else target * 2 + 1
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/form/ViewBlockAlignment.kt:49]: add empty line

File: galite-core/src/main/kotlin/org/kopi/galite/list/VIntegerCodeColumn.kt [hunk 4]
```diff
@@ -23,11 +23,11 @@ import kotlin.reflect.KClass
...
-  column: String,
-  names: Array<String>,
-  private val codes: Array<Int>,
-  sortAscending: Boolean
+        title: String,
+        column: String?,
+        names: Array<String>,
+        private val codes: Array<Int?>,
+        sortAscending: Boolean
 ) : VCodeColumn(title, column, names, sortAscending) {
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/list/VIntegerCodeColumn.kt:31]: check indentation

File: galite-core/src/main/kotlin/org/kopi/galite/form/BlockAlignment.kt [hunk 5]
```diff
@@ -15,7 +15,46 @@
...
+import java.io.Serializable
+
+/**
+ * A class to specify alignment in Layout
+ * Creates a new `BlockAlignment` instance.
+ *
+ * @param          ori          Represents the original block to be aligned with.
+ * @param          targets      Represents the alignment targets.
+ */
+class BlockAlignment(var ori : VBlock?,
```
[the remark this answers is not in the document]
Comment (SNsir) [reply galite-core/src/main/kotlin/org/kopi/galite/form/BlockAlignment.kt:30]: ignore it , to merge with master

File: galite-core/src/main/kotlin/org/kopi/galite/form/BlockAlignment.kt [hunk 6]
```diff
@@ -15,7 +15,46 @@
...
+    } else {
+      targets[x]
+    }
+  }
+
+  companion object {
+    const val ALG_LEFT = false
+    const val ALG_RIGHT = true
+  }
+}
```
[the remark this answers is not in the document]
Comment (SNsir) [reply galite-core/src/main/kotlin/org/kopi/galite/form/BlockAlignment.kt:60]: to merge with master

File: galite-core/src/main/kotlin/org/kopi/galite/form/VActorField.kt [hunk 7]
```diff
@@ -56,7 +56,7 @@ class VActorField(width: Int, height: Int) : VField(1, 1) {
 
   override fun getObjectImpl(r: Int): Any? = null
 
-  override fun toText(o: Any): String? = null
+  override fun toText(o: Any?): String? = null
```
[the remark this answers is not in the document]
Comment (SNsir) [reply galite-core/src/main/kotlin/org/kopi/galite/form/VActorField.kt:59]: ignore it , to merge with master

File: galite-core/src/main/kotlin/org/kopi/galite/form/VIntegerField.kt [hunk 8]
```diff
@@ -15,6 +15,370 @@
...
+  fun isNumeric(): Boolean = true
+
+  // ----------------------------------------------------------------------
+  // Interface Display
+  // ----------------------------------------------------------------------
+
+  /**
+   * return a list column for list
+   */
+  override fun getListColumn(): VListColumn {
```
[the remark this answers is not in the document]
Comment (SNsir) [reply galite-core/src/main/kotlin/org/kopi/galite/form/VIntegerField.kt:81]: inherited visibility used

[the remark this answers is not in the document]
Comment (h-haddad) [reply galite-core/src/main/kotlin/org/kopi/galite/form/VIntegerField.kt]: this function in VField should be protected : 
kopi :` protected abstract VListColumn getListColumn();`

File: galite-core/src/main/kotlin/org/kopi/galite/form/VIntegerField.kt [hunk 9]
```diff
@@ -201,7 +198,7 @@ class VIntegerField(width: Int,
   override fun getObjectImpl(r: Int): Any? = value[r]
 
   override fun toText(o: Any?): String {
-    return if (o == null) "" else (o as Int?).toString()
+    return o?.toString() ?: ""
   }
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VIntegerField.kt]: to single expression function

**Review by h-haddad — APPROVED** [review]

**Pull request merged**